src/main/scala/fm/xml/XmlReaderPath.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2018 Frugal Mechanic (http://frugalmechanic.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package fm.xml
+
+import fm.common.Implicits._
+import fm.common.Logging
+import javax.xml.bind.{JAXBContext, Unmarshaller}
+import javax.xml.stream.XMLStreamConstants.START_ELEMENT
+import org.codehaus.stax2.XMLStreamReader2
+import scala.collection.mutable
+import scala.reflect.{ClassTag, classTag}
+
+object XmlReaderPath {
+  // Apply helper for reading a non-mapped value
+  def apply[XmlValue: ClassTag](itemPath: String): XmlReaderPath[XmlValue, XmlValue] = XmlReaderPath[XmlValue, XmlValue](itemPath, identity)
+}
+
+/**
+  *
+  * @param itemPath The XPath-like path to the element we are interested in.  (e.g. "part", "items/part", etc)
+  * @param toMappedValue Map the XmlValue into another return type, this should match the return type in MultiXmlReader
+  * @tparam XmlValue The unmarshalled element class type
+  * @tparam MappedXmlValue A mapped XmlValue type
+  */
+final case class XmlReaderPath[XmlValue: ClassTag, MappedXmlValue](
+  itemPath: String,
+  toMappedValue: XmlValue => MappedXmlValue
+) extends Logging {
+  // Make this public, so we can use it for the XmlWriter JAXBContext
+  val itemClass: Class[XmlValue] = classTag[XmlValue].runtimeClass.asInstanceOf[Class[XmlValue]]
+  private[this] val jaxbContext: JAXBContext = JAXBContext.newInstance(itemClass)
+  private[this] val unmarshaller: Unmarshaller = jaxbContext.createUnmarshaller()
+
+  // The XPath-like path to the element we are interested in
+  // part => Array("part"), items/part => Array("items","part")
+  // This is reversed for easy comparison to the ArrayStack we use
+  private[xml] val path: Array[String] = itemPath.split('/')
+
+  // The name of the element we care about (last part of the path)
+  // items/part => part
+  private[xml] val itemName: String = path.last
+
+  private[xml] val targetDepth: Int = path.length - 1
+
+  private[xml] def readValue(xmlStreamReader: XMLStreamReader2): MappedXmlValue =  {
+    xmlStreamReader.require(START_ELEMENT, null, itemName)
+
+    val value: XmlValue = unmarshaller.unmarshal(xmlStreamReader, itemClass).getValue
+    toMappedValue(value)
+  }
+
+  // Check to see if the target depth and path match
+  private[xml] def elementMatches(xmlStreamReader: XMLStreamReader2, currentDepth: Int): Boolean = {
```

Comment:
Should probably needs more documentation as to how it is different from `pathMatches`